lib/reports/content_change_email_status_count.rb

```diff
@@ -0,0 +1,49 @@
+module Reports
+  class ContentChangeEmailStatusCount
+    def initialize(content_change)
+      @content_change = content_change
+      @email_statuses = email_statuses
+      @email_status_counts = email_status_counts
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      puts <<~TEXT
+        -------------------------------------------
+        Email status counts for Content Change #{@content_change.id}
+        -------------------------------------------
+
+        Sent emails: #{@email_status_counts[:sent]}
+
+        Pending emails: #{@email_status_counts[:pending]}
+
+        Failed emails: #{@email_status_counts[:failed]}
+
+        -------------------------------------------
+      TEXT
+    end
+
+  private
+
+    def email_statuses
+      subscription_contents_ids = @content_change.subscription_contents.pluck(:id)
+      email_ids = SubscriptionContent.where(id: subscription_contents_ids).pluck(:email_id)
+      Email.where(id: email_ids).pluck(:status)
```

Comment:
I think it might be worth doing `Email.where(id: email_ids).group(:status).count` here which I think will get you the same result as `email_status_counts` currently does.

Reply:
Very nice, thanks